sklearn/utils/validation.py

```diff
@@ -1877,16 +1877,12 @@ def _get_feature_names(X):
 
     types = sorted(t.__qualname__ for t in set(type(v) for v in feature_names))
 
-    # Warn when types are mixed and string is one of the types
+    # mixed type of string and non-string is not supported
```

Comment:
This may make it a bit more explicit.

```suggestion
    # Only string feature names are supported
```

Reply:
The thing is that this error is only raised when there's a mixed type.
I don't know the reason but according to the code, when there's a single type that is not ``str``, there's no error and it just returns None.